### PR TITLE
Add e2e tests for shoot-oidc-service pull requests

### PR DESCRIPTION
**Pull request by dimityrmirchev**
/kind enhancement

**What this PR does / why we need it**:
Add e2e tests execution step on pull requests towards `gardener-extension-shoot-oidc-service`.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: config/jobs/extension-shoot-oidc-service/extension-shoot-oidc-service-e2e-kind.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
+presubmits:
```
Comment (timebertt) [inline config/jobs/extension-shoot-oidc-service/extension-shoot-oidc-service-e2e-kind.yaml:1]: Side-note: later on, you might want to add a periodic job on master as well just like in g/g. :)

**Review by timebertt — COMMENTED** [review]
/lgtm

File: config/jobs/extension-shoot-oidc-service/extension-shoot-oidc-service-e2e-kind.yaml (hunk 1 above)
Comment (dimityrmirchev) [reply config/jobs/extension-shoot-oidc-service/extension-shoot-oidc-service-e2e-kind.yaml:1]: Yes, the step I added is a dummy step that is not yet implemented (I have a poc ready).
https://github.com/gardener/gardener-extension-shoot-oidc-service/blob/master/hack/test-e2e-provider-local.sh
I will try it in a PR and if everything is OK I will also submit a periodic job.

**Pull request merged**